src/assets.rs

```diff
@@ -145,9 +145,9 @@ impl HighlightingAssets {
     pub fn syntax_for_file_name(
         &self,
         file_name: impl AsRef<Path>,
-        mapping: &SyntaxMapping,
+        config: &Config,
```

Comment:
First of all, thanks a lot for working on this problem!

I don't have time for a detailed review right now, but that would be too early anyway. First we should look at the high level details. Sorry in advance if I say things that is the opposite if what I have said before.

I wonder if the code would not become nicer if we added the concept of "ignored suffixes" to SyntaxMapping. I think it conceptually fits quite nice there, because one can think of that we map "foo.rs~" to "foo.rs".

That would also allow us to keep these function signature unchanged in the API.

Reply:
@Enselic thanks for the suggestion. I moved it to `SyntaxMapping` module.